otelcol/collector.go

```diff
@@ -181,6 +181,11 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 		return multierr.Combine(err, col.service.Shutdown(ctx))
 	}
 	col.setCollectorState(StateRunning)
+
+	if err = col.service.ConfigResolved(ctx, cfg.ToStringMap()); err != nil {
```

Comment:
Since this functionality will be used by the OpAMP extension, it is worth noting that the extension will probably have to be restarted every time the configuration changes. I think this is acceptable if we plan for the Collector to only restart the components affected by a configuration change.